core/blockchain/blockchain_gateway.py

```diff
@@ -1,164 +1,154 @@
+    For more specific instructions check travis.yaml to see how travis does it.
     """
 
     def __init__(self):
         """
-        TODO: Refactor dependencies
-        TODO: deal with config
-        TODO: `communication_manager` is only used in a subset of methods,
-        consider separating
+        Initialize state to an empty list. Everything else is left to configure().
         """
         self.state = []
+        self.event = Event()
 
     def configure(self, config_manager: object, communication_manager: object):
+        """
+        Add communication_manager.
+        Set up IPFS client via config_manager.
+        """
         self.communication_manager = communication_manager
         config = config_manager.get_config()
         self.host = config.get("BLOCKCHAIN", "host")
         self.ipfs_port = config.getint("BLOCKCHAIN", "ipfs_port")
         self.port = config.getint("BLOCKCHAIN", "http_port")
         self.timeout = config.getint("BLOCKCHAIN", "timeout")
-        self.client = None
-        try:
-            self.client = ipfsapi.connect(self.host, self.ipfs_port)
-        except Exception as e:
-            logging.info("IPFS daemon not started, got: {0}".format(e))
-
-    ##########################################################################
-    ###                          PROVIDER SECTION                          ###
-    ##########################################################################
-
-    async def start_listening(self, event_filter: Callable,
-                                timeout=10) -> list:
-        """
-        Starts an indefinite loop that listens for a specific event to occur.
-        Called in `filter_set`. Filters are some condition that must be
-        fulfilled on a per tx basis
-        `poll_interval` specifies the number of seconds to stall before each
-        poll
-        """
-        timeout = time.time() + timeout
-        while time.time() < timeout:
-            global_state_wrapper = get_global_state(self.host, self.port, self.timeout)
-            filtered_diffs = filter_diffs(global_state_wrapper, self.state, event_filter)
-            if filtered_diffs:
-                return filtered_diffs
-            await asyncio.sleep(self.timeout)
-
-    def filter_set(self, event_filter: Callable, handler: Callable) -> list:
-        """
-        Calls async method `start_listening` and called by various listening
-        methods
-        """
-        asyncio.set_event_loop(asyncio.new_event_loop())
-        loop = asyncio.get_event_loop()
-        try:
-            filtered_diffs = loop.run_until_complete(
-                self.start_listening(event_filter))
-            if filtered_diffs:
-                check = list(map(handler, filtered_diffs))
-            else:
-                check = None
-        finally:
-            loop.close()
-        return check
+        self.client = ipfsapi.connect(self.host, self.ipfs_port)
```

Comment:
"ipfsapi.connect" has retry logic. We shouldn't catch the error, because the error would just cause everything to fail later. If we can't connect to IPFS, then there's no point in continuing to let the service run.

Reply:
I see your point. But I still think we should log the exception from IPFS (even if this means just raising the exception after logging) since we generally log all exceptions in the service.

Reply:
I agree with Neel. Let's bubble up the exception from IPFS to our custom module. (And do logging in case of error)

Reply:
Done